### ISSUE TITLE
add uutils/coreutils `mv` into nushell

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#10446]
### Related problem

Following on the momentum of moving the `uu_cp` crate into nushell, we'd like to add the `uu_mv` create too, as `umv` for now. We can use this issue to discuss parameters and other topics.

Looking for volunteers. Reply below and we can assign the issue to you.

### Describe the solution you'd like

Use the `ucp` code as a guide and port `umv` into nushell.

**Comment by fdncred** [comment]
Any takers? My dream of 1 coreutils command per week is sadly failing.

We need https://github.com/uutils/coreutils/pull/5335 to land I think before we can land an integration with mv.

**Comment by dmatos2012** [comment]
It looks like its already merged, so I will give mv a shot. Basically same procedure as `ucp`:
* Implement first iteration as `umv`,
* Write extra tests from `uutils mv` to `umv` tests.
* Run for the 3 weeks or something that @fdncred would like probably

If we are happy:
* Rename `umv` to `mv` and then `mv` to `mv-old`
* After period of time, remove `mv-old` altogether.

If thats ok, I will start with rm @fdncred.

**Comment by fdncred** [comment]
Yup, I was thinking a very similar procedure as `ucp`. I've assigned it to you. Thanks so much for chipping in again!

**Comment by dmatos2012** [comment]
I think I am almost done for first iteration of `umv`, @fdncred, but first I have to get some changes to `uutils` approved first, see PRs  [dir into itsef bug](https://github.com/uutils/coreutils/pull/5429) and [making UpdateMode public](https://github.com/uutils/coreutils/pull/5428).

other than this, I think there are only about 2 `nushell mv` tests that have different behavior but the new `umv` should almost be done. But I have got less time in these upcoming 2 days to work on this, but should have initial PR by the weekend or beginning of next week.

 Just so that you know that I am working on the background on it and I havent disappeared :).

**Comment by fdncred** [comment]
Thanks @dmatos2012 for the update. Looks like one of your PRs has landed. I'm very excited to have `umv` in nushell. Thanks for working on it! Our next release is Nov-14, so we have some time to work out the kinks.
